lib/nghttp2_session.c

```diff
@@ -3163,7 +3163,7 @@ ssize_t nghttp2_session_mem_send(nghttp2_session *session,
 }
 
 int nghttp2_session_send(nghttp2_session *session) {
-  const uint8_t *data;
+  const uint8_t *data = NULL;
```

Comment:
Thanks.  It might be explicit to set data to NULL here and nghttp2_session_mem_send instead of nghttp2_session_mem_send_internal, which would resolve your concern.
If you are OK, I'll pull this PR and do some amends and push it to the master branch.